### PR TITLE
force Night Mapping theme on Mac Mojave with Dark Theme

### PR DESCRIPTION
**Pull request by 3nids**
otherwise QGIS is pretty unusable
see #20195

### CODE REVIEW

**Comment by 3nids** [comment]
this should be seen seen as a temporary fix since Qt 5.12 should solve this issue
see https://bugreports.qt.io/browse/QTBUG-68891

before
image
after
image

**Comment by nyalldawson** [comment]
Shouldn't this be #ifed out to qt < 5.12 only? If upstream have a proper fix in 5.12 then we don't want to force night mapping theme there (since it has issues of its own).

**Comment by 3nids** [comment]
I'm not really sure.
Maybe a better approach would be to have 3 modes:
- auto (based on OS)
- default (=light)
- night (=dark)

That's maybe better and more future proof.

**Comment by 3nids** [comment]
I added `QgisApp::migrateOldSettings` because I did not find a place where there was something similar... did I miss it?

**Comment by 3nids** [comment]
works for you like this?

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -3200,7 +3200,21 @@ void QgisApp::setTheme( const QString &themeName )
   for the user to choose from.
   */
 
-  QgsApplication::setUITheme( themeName );
+  QString theme = themeName;
+#ifdef Q_OS_MAC
+#if QT_VERSION <= QT_VERSION_CHECK( 5, 12, 0 )
```
Comment (3nids) [inline src/app/qgisapp.cpp]:
```suggestion
#if QT_VERSION < QT_VERSION_CHECK( 5, 12, 0 )
```

File: src/native/qgsnative.h [hunk 2]
```diff
@@ -135,6 +135,11 @@ class NATIVE_EXPORT QgsNative : public QObject
      */
     virtual void setApplicationBadgeCount( int count );
 
+    /**
+     * This is used to force the use of the dark theme
```
Comment (nyalldawson) [inline src/native/qgsnative.h]: I'd just suggest updating the this to

"Returns true if the operating system is set to utilise a "dark" theme."

**Pull request merged**